src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Frameworks/SupportedTargetFrameworkAliasProvider.cs

```diff
@@ -0,0 +1,127 @@
+{
+    /// <summary>
+    ///     Responsible for producing valid values for the TargetFramework property from a design-time build.
+    /// </summary>
+    [ExportDynamicEnumValuesProvider("SupportedTargetFrameworkAliasEnumProvider")]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal class SupportedTargetFrameworkAliasProvider : ChainedProjectValueDataSourceBase<EnumCollection>, IDynamicEnumValuesProvider, IDynamicEnumValuesGenerator
+    {
+        private readonly IProjectSubscriptionService _subscriptionService;
+
+        [ImportingConstructor]
+        public SupportedTargetFrameworkAliasProvider(
+            ConfiguredProject project,
+            IProjectSubscriptionService subscriptionService)
+            : base(project, synchronousDisposal: true, registerDataSource: false)
+        {
+            _subscriptionService = subscriptionService;
+
+            ReadyToBuild = new OrderPrecedenceImportCollection<IConfiguredProjectReadyToBuild>(projectCapabilityCheckProvider: project);
+        }
+
+        [ImportMany]
+        public OrderPrecedenceImportCollection<IConfiguredProjectReadyToBuild> ReadyToBuild
+        {
+            get;
+        }
+
+        [ConfiguredProjectAutoLoad]
+        [AppliesTo(ProjectCapability.DotNet)]
+        public void Load()
+        {
+            // To avoid UI delays when opening the AppDesigner for the first time, 
+            // we auto-load so that we are included in the first design-time build
+            // for the project.
+            EnsureInitialized();
+        }
+
+        protected override IDisposable? LinkExternalInput(ITargetBlock<EnumCollectionProjectValue> targetBlock)
+        {
+            IProjectValueDataSource<IProjectSubscriptionUpdate> source = _subscriptionService.ProjectBuildRuleSource;
+
+            // Transform the changes from design-time build -> Supported target frameworks
+            DisposableValue<ISourceBlock<EnumCollectionProjectValue>> transformBlock = source.SourceBlock.TransformWithNoDelta(
+                update => update.Derive(Transform),
+                suppressVersionOnlyUpdates: false,
+                ruleNames: SupportedTargetFrameworkAlias.SchemaName);
+
+            // Set the link up so that we publish changes to target block
+            transformBlock.Value.LinkTo(targetBlock, DataflowOption.PropagateCompletion);
+
+            // Join the source blocks, so if they need to switch to UI thread to complete 
+            // and someone is blocked on us on the same thread, the call proceeds
+            JoinUpstreamDataSources(source);
+
+            return transformBlock;
+        }
+
+        private static EnumCollection Transform(IProjectSubscriptionUpdate input)
+        {
+            IProjectRuleSnapshot snapshot = input.CurrentState[SupportedTargetFrameworkAlias.SchemaName];
+
+            return snapshot.Items.Select(ToEnumValue)
+                                    .ToList();
```

Comment:
Do we want to sort these values, as done recently elsewhere?

Reply:
I think these are already sorted out:
![image](https://user-images.githubusercontent.com/8518253/125699653-cf7c7dfa-d3bc-4781-96ca-1c4207b97c0d.png)
I'll merge them as they are for now.